### PR TITLE
Remove `RewardsManager.sol`

### PR DESCRIPTION
**Pull request by MikeHathaway**
## Description

* Removed `RewardsManager.sol` and all respective tests
* Unwound RewardsManager invariants with PositionManager invariants

## Purpose

During our Sherlock and Kiril audits there was a bug discovered in `RewardsManager.sol`. Although we had previously agreed not to deploy the contract prior to the audit, having a implementation example with a bug is a bad idea. We opted to remove `RewardsManager.sol` all together in the event someone used it.

## Tasks

- [x] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [x] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [x] Scope labels have been assigned as appropriate.
- [x] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

File: tests/forge/invariants/PositionsAndRewards/handlers/ERC20PoolPositionHandler.sol [hunk 1]
```diff
@@ -47,4 +57,87 @@ contract ERC20PoolPositionHandler is PositionPoolHandler, BaseERC20PoolHandler {
         _quote = TokenWithNDecimals(_pool.quoteTokenAddress());
         _collateral = TokenWithNDecimals(_pool.collateralAddress());
     }
+
+    function lenderKickAuction(
```
Comment (MikeHathaway) [inline tests/forge/invariants/PositionsAndRewards/handlers/ERC20PoolPositionHandler.sol]: Do we still need access to `lenderKick` and `takeOrSettle` in position if we're no longer handling rewards?

Comment (EdNoepel) [reply tests/forge/invariants/PositionsAndRewards/handlers/ERC20PoolPositionHandler.sol]: Unsure.  We have regression tests which were using these with no staked liquidity.  Want to ensure we don't eliminate any `PositionManager` invariant coverage.

Comment (ith-harvey) [reply tests/forge/invariants/PositionsAndRewards/handlers/ERC20PoolPositionHandler.sol]: I don't think `lenderKick()` and `takeOrSettle()` are necessary. They were originally added to check invariants against rewards manager. See the original PR here -> https://github.com/ajna-finance/contracts/commit/ac303b81ffd76bf4a3b0a17e43d92a1cd8be8e39

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by EdNoepel — CHANGES_REQUESTED** [review]
Commit 9fd555c48a6f51fce46f2ae8281f8f6cdf39486b removes `ERC20PoolPositionHandler` and regression tests, which I spent substantial effort making operational.

**Review by grandizzy — COMMENTED** [review]
I prefer to be consistent in naming, that is rename `Positions` directories in tests to `PositionManager` (to match `ERC20Pool` and `ERC721Pool` namings)

**Comment by ith-harvey** [comment]
> Commit [9fd555c](https://github.com/ajna-finance/contracts/commit/9fd555c48a6f51fce46f2ae8281f8f6cdf39486b) removes `ERC20PoolPositionHandler` and regression tests, which I spent substantial effort making operational.

Apologies @EdNoepel , didn't take the time to review "why" you made the changes. Your changes successfully resolved `test_regression_index_out_of_bounds()` and `test_regression_max_less_than_min()`. I added them again in this commit -> https://github.com/ajna-finance/contracts/pull/1041/commits/3a63048a77a381449fabc84c1175b6c8a677aeed

**Comment by ith-harvey** [comment]
> I prefer to be consistent in naming, that is rename `Positions` directories in tests to `PositionManager` (to match `ERC20Pool` and `ERC721Pool` namings)

Resolved -> https://github.com/ajna-finance/contracts/pull/1041/commits/3a63048a77a381449fabc84c1175b6c8a677aeed

**Review by EdNoepel — APPROVED** [review]
LGTM; much thanks.

**Review by grandizzy — APPROVED** [review]

**Review by prat-gpt — APPROVED** [review]
LGTM

**Review by dmitriia — COMMENTED** [review]
Ok

**Pull request merged**